### PR TITLE
[docs] How to write product configuration

### PR DESCRIPTION
**Pull request by adydychk**

### CODE REVIEW

File: conf_how_to.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+"""
+DEVTOOLSET = 'source /opt/rh/devtoolset-6/enable'
+action('compiler version', cmd=f'bash -c "{DEVTOOLSET} && gcc --version"')
+
+
+# ==============================================================================
+# Configuration: archiving
+# ==============================================================================
+"""
+After the build you can specify what to archive with help of variables:
```
[the remark this answers is not in the document]
Comment (adydychk) [reply conf_how_to.py]: What do you mean? Description of archiving?

File: conf_how_to.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+action('install', stage=Stage.INSTALL, cmd=f'make DESTDIR={DEFAULT_OPTIONS["INSTALL_DIR"]} install')
+
+"""
+If you want to execute multiple commands during one actions you can use `&&`:
+"""
+DEVTOOLSET = 'source /opt/rh/devtoolset-6/enable'
+action('compiler version', cmd=f'{DEVTOOLSET} && gcc --version')
+
+"""
+Or execute them as a list:
```
Comment (adydychk) [inline conf_how_to.py]: Delete it

File: conf_how_to.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    'gcc --version',
+]
+action('compiler version', cmd=COMMANDS)
+
+"""
+On Ubuntu we recommend to call `bash -c "<your commands>"` because default
+shell on Ubuntu is `dash` :(
+"""
+DEVTOOLSET = 'source /opt/rh/devtoolset-6/enable'
+action('compiler version', cmd=f'bash -c "{DEVTOOLSET} && gcc --version"')
```
Comment (adydychk) [inline conf_how_to.py]: Use `bash -c` in all cases

File: conf_how_to.py [hunk 4]
```diff
@@ -85,17 +85,16 @@
...
-    "REPOS_DIR": root_dir / "repos",
-    "REPOS_FORKED_DIR": root_dir / "repos_forked",
-    "BUILD_DIR": root_dir / "build",
-    "INSTALL_DIR": root_dir / "install",
-    "PACK_DIR": root_dir / "pack",
-    "LOGS_DIR": root_dir / "logs",
-    "BUILD_TYPE": build_type,  # sets from command line argument ('release' by default)
-    "CPU_CORES": multiprocessing.cpu_count()  # count of logical CPU cores
-}
+`DEFAULT_OPTIONS` properties:
```
Comment (azhogov) [inline conf_how_to.py]: is dictionary that contains the following properties:

**Pull request merged**